src/com/maddyhome/idea/vim/action/change/delete/DeleteJoinVisualLinesAction.java

```diff
@@ -65,6 +69,6 @@ protected boolean execute(@NotNull Editor editor, @NotNull DataContext context,
 
   @Override
   public int getFlags() {
-    return Command.FLAG_EXIT_VISUAL;
+    return 0;
```

Comment:
Don't we need to return to the normal mode and reset visual selection after we have joined the selected lines? When I run `gJ`, I expect my selection to be empty, and it isn't now.